### PR TITLE
Bluetooth: Mesh: Further qualification improvements

### PR DESCRIPTION
**Pull request by jhedberg**
With these changes it's possible to pass most Network and Transport Layer test cases. Due to the impact on qualification, I recommend including these for 1.10.

### CODE REVIEW

**Comment by jhedberg** [comment]
@rymanluk @michal-narajowski FYI

File: subsys/bluetooth/host/mesh/shell.c [hunk 1]
```diff
@@ -959,6 +959,99 @@ static int cmd_mod_sub_del(int argc, char *argv[])
...
+
+	elem_addr = strtoul(argv[1], NULL, 0);
+
+	len = hex2bin(argv[2], label, sizeof(label));
+	memset(label + len, 0, sizeof(label) - len);
+
+	mod_id = strtoul(argv[3], NULL, 0);
+
+	if (argc > 4) {
+		cid = strtoul(argv[3], NULL, 0);
```
Comment (sjanc) [inline subsys/bluetooth/host/mesh/shell.c]: should this be argv[4] ?

File: subsys/bluetooth/host/mesh/shell.c [hunk 2]
```diff
@@ -959,6 +959,99 @@ static int cmd_mod_sub_del(int argc, char *argv[])
...
```
Comment (sjanc) [inline subsys/bluetooth/host/mesh/shell.c]: ditto

**Review by sjanc — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/mesh/shell.c (hunk 1 above)
Comment (jhedberg) [reply subsys/bluetooth/host/mesh/shell.c]: fixed

File: subsys/bluetooth/host/mesh/shell.c (hunk 2 above)
Comment (jhedberg) [reply subsys/bluetooth/host/mesh/shell.c]: fixed

**Review by sjanc — APPROVED** [review]

**Review by Vudentz — APPROVED** [review]
LGTM

**Pull request merged**